util/jsonutil/marshal.go

```diff
@@ -42,21 +44,94 @@ func MarshalTimestamp(t int64, stream *jsoniter.Stream) {
 	}
 }
 
-// MarshalValue marshals a point value using the passed jsoniter stream.
-func MarshalValue(v float64, stream *jsoniter.Stream) {
+// MarshalFloat marshals a float value using the passed jsoniter stream.
+func MarshalFloat(f float64, stream *jsoniter.Stream) {
 	stream.WriteRaw(`"`)
 	// Taken from https://github.com/json-iterator/go/blob/master/stream_float.go#L71 as a workaround
 	// to https://github.com/json-iterator/go/issues/365 (jsoniter, to follow json standard, doesn't allow inf/nan).
 	buf := stream.Buffer()
-	abs := math.Abs(v)
+	abs := math.Abs(f)
 	fmt := byte('f')
 	// Note: Must use float32 comparisons for underlying float32 value to get precise cutoffs right.
 	if abs != 0 {
 		if abs < 1e-6 || abs >= 1e21 {
 			fmt = 'e'
 		}
 	}
-	buf = strconv.AppendFloat(buf, v, fmt, -1, 64)
+	buf = strconv.AppendFloat(buf, f, fmt, -1, 64)
 	stream.SetBuffer(buf)
 	stream.WriteRaw(`"`)
 }
+
+// MarshalHistogram marshals a histogram value using the passed jsoniter stream.
+// It writes something like:
+//
+//	{
+//	    "count": "42",
+//	    "sum": "34593.34",
+//	    "buckets": [
+//	      [ 3, "-0.25", "0.25", "3"],
+//	      [ 0, "0.25", "0.5", "12"],
+//	      [ 0, "0.5", "1", "21"],
+//	      [ 0, "2", "4", "6"]
+//	    ]
+//	}
+//
+// The 1st element in each bucket array determines if the boundaries are
+// inclusive (AKA closed) or exclusive (AKA open):
+//
+//	0: lower exclusive, upper inclusive
+//	1: lower inclusive, upper exclusive
+//	2: both exclusive
+//	3: both inclusive
+//
+// The 2nd and 3rd elements are the lower and upper boundary. The 4th element is
+// the bucket count.
+func MarshalHistogram(h *histogram.FloatHistogram, stream *jsoniter.Stream) {
```

Comment:
I believe this is copied from api.go, so I did not review it

Reply:
Yes, I believe the same. :)